### DISCUSSION TITLE
Implementing WiFiBroadcast with MotionEye or Similar

### DISCUSSION THREAD

**Discussion by thekodiak13** [svpcom/wfb-ng#181] [Questions]
Hey -- first of all, great work on this project. You're a very talented developer.

I'd like to try to utilize this project in a little different way. I own some land with multiple cabins, as well as my own house. I would like to set up a security  system that has a camera at each cabin that all feed to a receiver HUB at my house. I would like to have these streams input into something like [MotionEye](https://github.com/ccrisan/motioneye) or equivalent (for motion alerts, recording, etc). These cabins do not have internet access and I want to avoid using cellular ($$$). Each camera would need a unique IP Address or port in order for me to add them into MotionEye.

Is that something that would be possible using this project?

Thanks for your time!

**Comment by thekodiak13** [comment]
Maybe another way I could do it is setting up a LAN<--->LAN setup. So have a Pi on the transmit end with a camera and WiFi radio, and a Pi+WiFi radio on the receiver end.

Is a LAN<--->LAN type setup achievable using WiFiBroadcast? And would it be capable of streaming HD video?

Happy to draw a diagram of what I'm trying to accomplish if the above description doesn't make sense.

Thank you!

**Reply by svpcom** [reply]
Why ordinary wifi link doesn't meed your needs?

**Reply by thekodiak13** [reply]
Mostly distance and interference reasons. Some cabins are over a kilometer from where my main hub would be.

**Reply by svpcom** [reply]
WFB will help with distance, but not with interference.

**Reply by thekodiak13** [reply]
Understood. I still would like to attempt this if you think it is possible....?

**Reply by svpcom** [reply]
Yes it possible, but it requires directed antennas for this distance.

**Reply by thekodiak13** [reply]
Oh ok, so some Alfa AWUS036ACH's won't work? I'll look into directional antennas (any recommendations?).

Once I have wifibroadcast and all dependencies up and running, what do you suggest for configuration options? I assume I'll need to change some things in `telemetry/conf/master.cfg` but I'm a little lost at what I need to change.

**Reply by svpcom** [reply]
You need to write gstreamer pipeline which will capture rtsp stream from camera and convert it to rtp udp stream
For example:
```
gst-launch-1.0 rtspsrc protocols=tcp location=rtsp://user:password@1.2.3.4:554/Streaming/Channels/101 ! udpsink host=127.0.0.1 port=5600 sync=false
```
On the RX side you need to convert RTP stream to needed format.
For example (will show h265 video in a window):
```
gst-launch-1.0  -v udpsrc port=5600 caps='application/x-rtp, media=(string)video, encoding-name=(string)H265' ! rtph265depay ! avdec_h265 ! xvimagesink sync=false
```

**Reply by svpcom** [reply]
Each camera model has own format for RTSP url, so you need to find it in camera manual

**Reply by svpcom** [reply]
For antenna setup you need to read ez-wifibroadcast and/or OpenHD forums

**Reply by thekodiak13** [reply]
Thanks a lot buddy. I'll work on getting this set up. I've gone through EZ-Wifibroadcast and OpenHD but liked your project because of the encryption. If I'm able to get this to work, I'll report back.

Thanks for your time!
